nilearn/glm/first_level/first_level.py

```diff
@@ -513,6 +514,26 @@ def _check_fit_inputs(
         if not isinstance(run_imgs, (list, tuple)):
             run_imgs = [run_imgs]
 
+        # check that mask type and image types are compatible
```

Comment:
I suspect that this will probably need to be extracted into a separate function to be reused for first or second level (and also some decoders)?